app/views/SelectServerView.tsx

```diff
@@ -12,72 +12,60 @@ import database from '../lib/database';
 
-interface ISelectServerViewState {
-	servers: TServerModel[];
-}
+const SelectServerView = () => {
+	const [servers, setServers] = React.useState<TServerModel[]>([]);
 
-interface ISelectServerViewProps {
-	navigation: StackNavigationProp<ShareInsideStackParamList, 'SelectServerView'>;
-	server?: string;
-}
+	const server = useAppSelector(state => state.share.server.server);
+	const navigation = useNavigation<StackNavigationProp<ShareInsideStackParamList, 'SelectServerView'>>();
 
-class SelectServerView extends React.Component<ISelectServerViewProps, ISelectServerViewState> {
-	static navigationOptions = (): StackNavigationOptions => ({
-		title: I18n.t('Select_Server')
-	});
+	useLayoutEffect(() => {
+		navigation.setOptions({
+			title: I18n.t('Select_Server')
+		});
+	}, [navigation]);
 
-	state = { servers: [] };
-
-	async componentDidMount() {
-		const serversDB = database.servers;
-		const serversCollection = serversDB.get('servers');
-		const servers = await serversCollection.query(Q.where('rooms_updated_at', Q.notEq(null))).fetch();
-		this.setState({ servers });
-	}
-
-	select = async (server: string) => {
-		const { server: currentServer, navigation } = this.props;
+	useEffect(() => {
+		const init = async () => {
+			const serversDB = database.servers;
+			const serversCollection = serversDB.get('servers');
+			const serversResult = await serversCollection.query(Q.where('rooms_updated_at', Q.notEq(null))).fetch();
+			setServers(serversResult);
+		};
+		init();
+	}, []);
 
+	const select = async (serverSelected: string) => {
 		navigation.navigate('ShareListView');
-		if (currentServer !== server) {
-			await shareExtensionInit(server);
+		if (serverSelected !== server) {
+			await shareExtensionInit(serverSelected);
 		}
 	};
 
-	renderItem = ({ item }: { item: TServerModel }) => {
-		const { server } = this.props;
-		return <ServerItem onPress={() => this.select(item.id)} item={item} hasCheck={item.id === server} />;
-	};
-
-	render() {
-		const { servers } = this.state;
-		return (
-			<SafeAreaView>
-				<StatusBar />
-				<FlatList
-					data={servers}
-					keyExtractor={keyExtractor}
-					renderItem={this.renderItem}
-					getItemLayout={getItemLayout} // Refactor row_height
-					contentContainerStyle={List.styles.contentContainerStyleFlatList}
-					ItemSeparatorComponent={List.Separator}
-					ListHeaderComponent={List.Separator}
-					ListFooterComponent={List.Separator}
-					removeClippedSubviews
-					keyboardShouldPersistTaps='always'
-				/>
-			</SafeAreaView>
-		);
-	}
-}
-
-const mapStateToProps = ({ share }: IApplicationState) => ({
-	server: share.server.server
-});
+	return (
+		<SafeAreaView>
+			<StatusBar />
+			<FlatList
+				data={servers}
+				renderItem={({ item }: { item: TServerModel }) => (
+					<ServerItem onPress={() => select(item.id)} item={item} hasCheck={item.id === server} />
+				)}
+				keyExtractor={keyExtractor}
+				getItemLayout={getItemLayout} // Refactor row_height
```

Comment:
```suggestion
				getItemLayout={getItemLayout}
```

Reply:
This comment will continue to remind us to refactor this getItemLayout, because he is using a fixed height and could mess up the screen with a more oversized letter.